frame/staking/src/migrations.rs

```diff
@@ -83,9 +83,6 @@ pub mod v7 {
 		let validator_count = Validators::<T>::iter().count() as u32;
 		let nominator_count = Nominators::<T>::iter().count() as u32;
 
-		CounterForValidators::<T>::put(validator_count);
```

Comment:
This is technically making this migration code _WRONG_. Either remove it completely, or make sure it is correct. 

we don't enforce this now, but a correct way would be to make sure this migration code stays correct over time. That is, it should have its own storage items and not depend on the pallet types. In this case, you need to create (mock) storage types via `generate_storage_alias` and use them to set a correct value for the counters. 

Or, we need some special `set_counter` to set the counters upon migration, but I don't think we have this for now. 

If all of this is too much, with despair and sadness, I am also fine with just removing the migration code altogether. But the long term correct thing to do is as I said above.

Reply:
Could you help with the correct implementation details on how to write the migration code, I am still new to the Substrate repo.

Reply:
you can look into other use cases of `generate_storage_alias` and its documentation. This allows you to generate a storage value type that mimics the one that is being removed. In this case, you can create a 

```
generate_storage_value!(CounterForValidators => Value<u32>);
```

and now you have `CounterForValidators` back in scope, and you can use it.

Reply:
is staking pallet using the pallet macro in v7 ?
If so then the generate_storage_value can use a wrong pallet prefix.

Reply:
Is this question for me or @kianenigma

Reply:
yeah it was already migrated. Then the very pedantic course of action would be to accept the prefix as the argument of the migration function, but I will be fine with skipping it since the migration code is already at risk to becoming outdated (since it depends on `T: Config`). So the assumption is that whoever uses this migration code should test it, and we're good.